CHANGELOG.md

```diff
@@ -29,12 +29,17 @@
 * Fix a race condition which would lead to "uncaught exception in notifier thread: N5realm15InvalidTableRefE: transaction_ended" and a crash when the source Realm was closed or invalidated at a very specific time during the first run of a collection notifier (Core upgrade)
 
 ### Enhancements
+* Add support for the `GUID` data type. It can be used as primary key and is indexable. (PR [#2120](https://github.com/realm/realm-dotnet/pull/2120))
+* Replaced the implementation of the string query parser (the one used for [`realm.All().Filter("some-string-query")`](https://docs.mongodb.com/realm-sdks/dotnet/10.0.0-beta.3/reference/Realms.CollectionExtensions.html#Realms_CollectionExtensions_Filter__1_System_Linq_IQueryable___0__System_String_)). This results in ~5% reduction of the size of the native binary while keeping the query execution times on par with the old parser. (PR [#2185](https://github.com/realm/realm-dotnet/pull/2185), Core upgrade)
+* Added support for value substitution in string based queries. So [`realm.All<T>().Filter("field1 = $0 && field2 = $1", value1, value2)`](https://github.com/realm/realm-js/blob/master/docs/tutorials/query-language.md) is now possible
```

Comment:
I don't know why what's above my line looks like added by me, but I clearly only added this third point in this pr.

Reply:
Mmh, I think it happened to me too at some point. Maybe you did not branche from master when you created this branch?

Reply:
No I did not branch from master, but from the bison branch, since I needed that.
What's strange is that if you look at the commits, they count also the once coming from where I branched. I am not sure why that is. It leaves me a bit confused.

Reply:
I would suggest doing interactive rebase on top of master as opposed to doing a merge. The issue is that the branch you branched off of got squashed into `master`, which means that it now sees two divergent sequences of changes that are in fact identical.

Reply:
Oh, you're totally right. Merge was the mistake, rebase is what I should have done. Note taken for the next time.